harness/apisupport.harness/nbproject/project.properties

```diff
@@ -16,7 +16,8 @@
 # under the License.
 
 javac.compilerargs=-Xlint -Xlint:-serial
-javac.source=1.6
+javac.source=1.8
+javac.target=1.8
```

Comment:
You don't need to set `javac.target` - @JaroslavTulach ensured this in https://github.com/apache/netbeans/commit/4661771. The build system ensures, that `javac.target` is set to the value of `javac.source` is the former is not set.

Reply:
ok. but it doesn't hurt right? does everything use the same build xml which has this functionality? having source and target at different values can cause trouble if its missed somewhere. (ultimately we should aim to use --release in distant future)

i can remove it if you prefer it gone.

Reply:
You can see the mechanic in the `build.xml`:

https://github.com/apache/netbeans/blob/61795d673369f587d2ae2f50254b442a407b64c0/ide/css.editor/build.xml#L22-L25

The build looks simple, but truth is, that it just inherits its power from:

-  `nbbuild/templates/projectized.xml` (direct)
-  `nbbuild/templates/common.xml` (transitive)
-  `nbbuild/default.xml`
-  `nbbuild/default-properties.xml`
-  `nbbuild/jdk.xml`
-  `nbbuild/binaries-default-properties.xml`

The `javac.source` and `javac.target` default values are set by the `basic-init` target in `nbbuild/templates/common.xml`. 

From the default target `build` it is reached via:

- `build`
- `netbeans
- `-netbeans`
- `init`
- `basic-init`

For using `--release`, this was also tackled by @JaroslavTulach (I think it is in the same PR) - see here:

https://github.com/apache/netbeans/blob/96720b9d39122e514fe21536a831247e5dc17b10/nbbuild/antsrc/org/netbeans/nbbuild/CustomJavac.java#L68-L91

This works only for normal module builds, special cases, where the ant `javac` task is called directly, adjustments need to be made there.

NetBeans build is a complex beast, but it some very nice things in it, at the price that its complex.

For the `javac.target`, yes I think it would be good to not add them, else next time, someone will argue, that the two should be adjusted independently from each other.

Reply:
thanks for the info. will remove it.

Reply:
done, removed `javac.target` from all properties I touched, even if it was there before.

I kept the changes to build xmls since they all had target specified.

Reply:
it doesn't build. looks like i have to add a few targets again.

Reply:
ok, it builds again on JDK 17.

please note that this branch is fairly old and i don't want to abuse the CI too much just to rebase and push again. So to get the JDK 17 gradle changes do a
`
git pull --rebase git@github.com:apache/netbeans.git master`

after you checked out this branch.